libs/s25main/Debug.cpp

```diff
@@ -62,16 +62,16 @@ bool captureBacktrace(std::vector<void*>& stacktrace, LPCONTEXT ctx = nullptr) n
     if(!kernel32 || !dbghelp)
         return false;
 
-#if __GNUC__ >= 9
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wcast-function-type"
-    // error: cast between incompatible function types from
-    // 'FARPROC' {aka 'int (__attribute__((stdcall)) *)()'}
-    // to
-    // 'RtlCaptureContextType' {aka 'void (__attribute__((stdcall)) *)(CONTEXT*)'}
-    // [-Werror=cast-function-type]
-    // and so on
-#endif
+#            if __GNUC__ >= 9
```

Comment:
I do agree this looks odd/ugly here but serves the point to highlight that it is in multiple conditions. And e.g. https://github.com/Return-To-The-Roots/s25client/pull/1582/files#diff-e93d2e7c03a97a61287ef951276877e6d0cacac56ce2e0daecc3c935bae907eaR24-R35 looks much better due to this.

I'm wondering if we couldn't use Boost for this (backtrace collection) anyway or at least add a wrapper around `GetProcAddress` to do the casting with the warning disabled or use Boost.DLL which IIRC can do that too.

But that's another PR...